### DISCUSSION TITLE
jupyternotebook 讀不到 .bit

### DISCUSSION THREAD

**Discussion by chengyu-123** [bol-edu/HLS-SOC-Discussions#68] [Q&A]
*********請益*********
.hwh, .bit 都有上傳到 jupyternotebook , 路徑也看起來是對的，還是吃不到 .bit 有可能是什麼原因~
<img width="174" alt="1" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/b44f4aa5-3456-436b-8a6f-67bcc0facaf9">
<img width="386" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/ebb8e543-0a09-412a-bb43-15d0d1bcfeb0">
<img width="569" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/42027ba4-d6e8-4aee-aaa0-98baba8ba00d">

**Comment by nthuyouwei** [comment]
你的.bit 跟.hwh 前面的檔名要一致！

henry ***@***.***>於 2023年9月24日 週日，上午12:38寫道：

**Reply by chengyu-123** [reply]
感謝提醒!!!
但改完後還是讀不到，換了幾個板子也無法，再請路高手幫我解惑一下~
<img width="225" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/8bafe3b0-3e47-4253-a5c3-b3412c436597">
<img width="444" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/4ee51db2-0643-48f0-8330-c14378a54d85">

**Comment by nthuyouwei** [comment]
你一開始是oserror(是檔名問題沒錯)，但你現在讀不到原因沒附上，如果是timeoutError，這應該是板子的關係，我剛剛試了一下06 11 01 17這四個都不行，但試到09就可以了！如是其他問題附上error資訊

henry ***@***.***>於 2023年9月24日 週日，上午1:11寫道：

**Comment by samlo123** [comment]
重新跑一次產生.bit檔，板子重新借（要選對是pynq or kv260），還有ol=Overlay的那行檔案位置會因板子而不同，試試看。
我跑第二次就好了。

**Comment by chengyu-123** [comment]
@nthuyouwei @samlo123   感謝二位的幫助!! 多試幾個板子就ok了~
@samlo123 想請教一下，為什麼"ol=Overlay的那行檔案位置會因板子而不同"呢，我以為只跟 Jupyter notebook 上傳 .bit & .hwh 位置有關

**Reply by samlo123** [reply]
lab2的pdf有說使用的板子不同，相關設定的會有差異。可以去翻看看。

**Reply by liyuan-chang** [reply]
可以在 Jupyter notebook 裡面用 `!pwd` 和 `!ls` 指令來確認檔案的路徑，pynq 的路徑是 `/home/xilinx/jupyter_notebooks/`，而 kv260 則是 `/home/root/jupyter_notebooks/`，如果 .bit, .hwh 和 .ipynb 都放在同一個資料夾，其實也可以直接使用相對路徑。
